packages/completer/src/manager.ts

```diff
@@ -24,6 +25,13 @@ export class CompletionProviderManager implements ICompletionProviderManager {
   constructor() {
     this._providers = new Map();
     this._panelHandlers = new Map();
+    this._providersActivated = new Signal<ICompletionProviderManager, void>(
+      this
+    );
+  }
+
+  get providersActivated(): ISignal<ICompletionProviderManager, void> {
```

Comment:
Design question, are you interested by listening to the changes in the active providers or in the readiness of the manager at load time?

If this is the second, I would use a `ready` attribute:

```ts
get ready(): Promise<void>
```

If not, the signal is good, I would use anoterh name like `activeProvidersChanged` to better reflect when the signal is emitted (as it will be emitted on settings update too).

> Look for example at `jupyterlab/jupyterlab/packages/apputils/src/sessioncontext.tsx`